asterisk_click2dial/models/asterisk_server.py

```diff
@@ -0,0 +1,283 @@
+                    )
+                )
+            for check_str in [dialplan_context, alert_info, login, password]:
+                if check_str[1]:
+                    try:
+                        check_str[1].encode("ascii")
+                    except UnicodeEncodeError:
+                        raise ValidationError(
+                            _(
+                                "The '%s' should only have ASCII caracters for "
+                                "the Asterisk server '%s'" % (check_str[0], server.name)
+                            )
+                        )
+
+    @api.model
+    def _connect_to_asterisk(self):
+        """
+        Open the connection to the Asterisk Manager
+        Returns an instance of the Asterisk Manager
+
+        """
+        user = self.env.user
+        ast_server = user.get_asterisk_server_from_user()
+        # We check if the current user has a chan type
+        if not user.asterisk_chan_type:
+            raise UserError(_("No channel type configured for the current user."))
+
+        # We check if the current user has an internal number
+        if not user.resource:
+            raise UserError(_("No resource name configured for the current user"))
+
+        _logger.debug("User's phone: %s/%s", user.asterisk_chan_type, user.resource)
+        _logger.debug("Asterisk server: %s:%d", ast_server.ip_address, ast_server.port)
+
+        # Connect to the Asterisk Manager Interface
+        try:
+            ast_manager = Manager.Manager(
+                (ast_server.ip_address, ast_server.port),
+                ast_server.login,
+                ast_server.password,
+            )
+        except Exception as e:
+            _logger.error(
+                "Error in the request to the Asterisk Manager Interface %s",
+                ast_server.ip_address,
+            )
+            _logger.error("Here is the error message: %s", e)
+            raise UserError(
+                _(
+                    "Problem in the request from Odoo to Asterisk. "
+                    "Here is the error message: %s" % e
+                )
+            )
+
+        return (user, ast_server, ast_manager)
+
+    def test_ami_connection(self):
+        self.ensure_one()
+        ast_manager = False
+        try:
+            ast_manager = Manager.Manager(
+                (self.ip_address, self.port), self.login, self.password
+            )
+        except Exception as e:
+            raise UserError(_("Connection Test Failed! The error message is: %s" % e))
+        finally:
+            if ast_manager:
+                ast_manager.Logoff()
+        raise UserError(
+            _(
+                "Connection Test Successfull! Odoo can successfully login to "
+                "the Asterisk Manager Interface."
+            )
+        )
+
+    @api.model
+    def _get_calling_number_from_channel(self, chan, user):
+        """Method designed to be inherited to work with
```

Comment:
Inconsistent use of `"""`, above the text was after a newline and the closing `"""`as well

Reply:
fixed